### PR TITLE
expose additionalDetails of imperative errors if any

### PR DESCRIPTION
**Pull request by CBforZ**
## Proposed changes

If an imperative error is thrown previously we were not  displaying its `additionalDetails` value 

initially we talked about seeing how big this change needed to be, and potentially scope it to only certain operations, but considering we weren't doing anything with additionalDetails I think this should work for most things  

https://github.com/zowe/zowex/issues/766

We had talked about targeting `main` with this PR but I don't know of a z/OSMF only error that will demo this whereas you can trigger the original  error in the above issue  to get the additional detail.  I can re-create the error if it makes more sense to target main but this is testable with the zowex code... let me know 

## Release Notes

Milestone:

Changelog:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**All checks must pass before this pull request is reviewed by the Zowe Explorer squad.**

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`
- [ ] New ZE APIs are tested with extender types that haven't adopted yet to determine breaking changes. Can use Zowe zFTP marketplace extension.

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have tested new functionality with the FTP extension and profile verifying no extender profile type breakages introduced
- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

## Further comments

image

### CODE REVIEW

**Review by traeok — COMMENTED** [review]
Thanks for working on this Chris, the changes make sense to me.

~~Before I approve, can you add one or more unit tests to get the patch coverage up to the minimum threshold? (90%) This should get the `patch` Codecov stage passing.~~

Please disregard the patch coverage, Codecov is not reporting accurate coverage metrics for the staging branch.
That said, I still think we could benefit from a unit test to make sure that additional details are included in the displayed message - not explicitly requesting changes for this since the changes work as-is.

**Review by traeok — APPROVED** [review]

**Comment by CBforZ** [comment]
Adding a test is a good idea, I'd just have to find if we have any examples of asserting on notifications

**Comment by CBforZ** [comment]
Added a test

**Review by pujal0909 — APPROVED** [review]

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 ### Bug fixes
 
+- Fixed an issue where additional error details were not displayed, particularly using SSH profiles: [zowex/#766](https://github.com/zowe/zowex/issues/766).
```
Comment (anaxceron) [inline packages/zowe-explorer/CHANGELOG.md]:
```suggestion
- Fixed an issue where additional error details were not displayed, particularly using SSH profiles. [zowex/#766](https://github.com/zowe/zowex/issues/766)
```

**Review by anaxceron — COMMENTED** [review]
looks good, just left a comment re: formatting changes.

**Review by anaxceron — APPROVED** [review]
🐅

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @CBforZ!
